docs/DeveloperGuide.md

```diff
@@ -1,38 +1,455 @@
+| v1.0    | Property Manager | check the details of a property       | view the property's information                                      |
+| v1.0    | Property Manager | pair a client to a property           | record down which client is renting which property                   |
+| v1.0    | Property Manager | unpair a client to a property         | update my rental records when a client is no longer renting property |
+| v1.0    | Property Manager | save my data                          | used the data created from a previous use of the app                 |
+| v1.0    | Property Manager | quit the app                          | -                                                                    |
+| v2.0    | Property Manager | check the details of a client         | view the client's information                                        |
+| v2.0    | Property Manager | search clients using their details    | easily find specific clients                                         |
+| v2.0    | Property Manager | search properties using their details | easily find specific properties                                      |
 
+___
+## Design
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Architecture
+![Software Architecture Diagram](diagrams/ArchitectureDiagram.png)
+### UI Component
+### Parser Component
+### Client Component
+### Property Component
+### List Component
+The list feature has the following commands in it -
+* `list -client` This lists every client, along with all their information
+* `list -property` This lists every property, along with all its information
+* `list -everything` This lists everything about both clients and properties
+* `list -client FLAG` This lists only the information present in the TAG for every client. The types of
+  TAGs are -
+    * `c/` This is for contact number
+    * `b/` This is for budget
+    * `n/` This is for name
+    * `e/` This is for e-mail
+* `list -property FLAG` This lists only the information present in the TAG for every property. The types
+  of TAG are -
+    * `a/` This is for address
+    * `n/` This is for name
+    * `p/` This is for price
+    * `t/` This is for type  
 
+There are 5 different classes, that each inherit from the abstract Command class. The commands read information from 
+the PropertyList and ClientList classes respectively, and display using the Ui class, making use of the objects of 
+these classes. The Commands which display all the information - i.e. CommandListClients, CommandListProperties, and
+CommandListEverything read and display using loops inside the overriden execute() method itself. The Commands which 
+display selected information - i.e. CommandListClientsWithTags and CommandListPropertiesWithTags use their private 
+methods to display their information. The class structure is as follows - 
+![ListClassDiagram](diagrams/ListClassDiagram.png)
```

Comment:
This diagram is too small. Please enlarge it. 😃

Also, you can remove the C and A logos beside the classes and abstractions.